tests/features/camel-hazelcast/src/test/java/org/apache/karaf/camel/itest/CamelHazelcastITest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.karaf.camel.itest;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.karaf.camel.itests.AbstractCamelSingleFeatureResultMockBasedRouteITest;
+import org.apache.karaf.camel.itests.AvailablePortProvider;
+import org.apache.karaf.camel.itests.CamelKarafTestHint;
+import org.apache.karaf.camel.itests.PaxExamWithExternalResource;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+
+@CamelKarafTestHint(externalResourceProvider = CamelHazelcastITest.ExternalResourceProviders.class)
+@RunWith(PaxExamWithExternalResource.class)
+@ExamReactorStrategy(PerClass.class)
+public class CamelHazelcastITest extends AbstractCamelSingleFeatureResultMockBasedRouteITest {
+
+    private static final int HZ_PORT = 5701;
+
+    @Override
+    public void configureMock(MockEndpoint mock) {
+        //Map + replicated Map + 2 for list + Queue + Set + Topic
+        mock.expectedBodiesReceived( Collections.nCopies(7, "OK"));
+    }
+
+    @Test
+    public void testResultMock() throws Exception {
+        assertMockEndpointsSatisfied();
+    }
+
+    @After
+    public void stopCamelContext() {
+        //to trigger graceful shutdown of hazelcast components before uninstalling bundles
+        getContext().stop();
+    }
```

Comment:
Should be moved to the integration test framework, don't you agree?

Reply:
Yes, done, I just had to override the `stopContext()`  for the camel-core because otherwise the context gets stopped to early (after the first test)

Reply:
Removed and used the `System.setProperty("hazelcast.shutdownhook.enabled","false");`